### PR TITLE
[prometheus-server] adding sidecar discovery

### PR DESCRIPTION
**Pull request by viennaa**

### CODE REVIEW

File: common/prometheus-server/templates/thanos/storeendpoint.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+{{- $root := . }}
+{{- range $name := coalesce .Values.names .Values.global.targets (list .Values.name) }}
+{{- range $thanos := $.Values.thanos.sidecarDiscovery }} 
+---
+apiVersion: monitoring.banzaicloud.io/v1alpha1
+kind: StoreEndpoint
+metadata:
+  name: {{ include "prometheus.name" (list $name $root) }}-to-{{ $thanos.thanosName }}
+spec:
+  url: {{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.kubernetes.{{- required ".Values.global.region missing" $root.Values.global.region -}}.{{- required ".Values.global.domain missing" $root.Values.global.domain }}:10901
```
Comment (richardtief) [inline common/prometheus-server/templates/thanos/storeendpoint.yaml:11]: Please add the default cluster domain as well: `cluster.local`. Or make it customisable via values.

**Review by richardtief — CHANGES_REQUESTED** [review]
Thanks for this additional feature. One litte remark.

File: common/prometheus-server/templates/thanos/storeendpoint.yaml (hunk 1 above)
Comment (richardtief) [reply common/prometheus-server/templates/thanos/storeendpoint.yaml:11]: We can also use this pattern w/o port:
`dnssrvnoa+_grpc._tcp.{{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.{{ .Values.clusterDomain }}`

**Review by richardtief — CHANGES_REQUESTED** [review]

File: common/prometheus-server/templates/thanos/storeendpoint.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+---
+apiVersion: monitoring.banzaicloud.io/v1alpha1
+kind: StoreEndpoint
+metadata:
+  name: {{ include "prometheus.name" (list $name $root) }}-to-{{ $thanos.thanosName }}
+spec:
+  {{- if $thanos.domain }}
+  url: {{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.kubernetes.{{ $thanos.domain }}:10901
+  {{- else }}
+  url: {{ include "prometheus.name" (list $name $root) }}.{{ $thanos.namespace }}.svc.kubernetes.{{- required ".Values.global.region missing" $root.Values.global.region -}}.{{- required ".Values.global.domain missing" $root.Values.global.domain }}:10901
```
[the remark this answers is not in the document]
Comment (viennaa) [reply common/prometheus-server/templates/thanos/storeendpoint.yaml]: we go with cluster.local.
Readme updated accordingly

**Review by richardtief — APPROVED** [review]
Thanks! Looks good.

**Pull request merged**